BacktraceScreen.h

```diff
@@ -0,0 +1,66 @@
+#ifndef HEADER_BacktraceScreen
+#define HEADER_BacktraceScreen
+/*
+htop - BacktraceScreen.h
+(C) 2024 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "Object.h"
+#include "Panel.h"
+#include "Process.h"
+#include "Settings.h"
+
+
+typedef struct BacktracePanelPrintingHelper_ {
+   size_t maxAddrLen;
+   size_t maxDemangledFuncNameLen;
+   size_t maxFuncNameLen;
+   size_t maxFrameNumLen;
+   size_t maxObjPathLen;
+   size_t maxObjNameLen;
+} BacktracePanelPrintingHelper;
+
+typedef enum BacktraceScreenDisplayOptions_ {
+   NO_OPTION = 0,
+   DEMANGLE_NAME_FUNCTION = 1 << 0,
+   SHOW_FULL_PATH_OBJECT = 1 << 1,
+} BacktraceScreenDisplayOptions;
```

Comment:
1. Maybe rename this type to `BacktracePanelDisplayOptions`.
2. Consider using a struct of booleans.

```c
typedef struct BacktracePanelDisplayOptions_ {
   bool showDemangledFuncName;
   bool showFullPathObject;
} BacktracePanelDisplayOptions;
```
The reason I suggest turning this into a boolean structure is that, there's probably no size saving for using a bitfield here. There might be just one instance of `BacktracePanel` at any time, and testing `(booleanVal & 1) != 0` might take slightly more code size than just `booleanVal != 0`.

Reply:
I'm fine both ways. If you use a struct, I'd recommend to use bitfields:

```c
typedef struct BacktracePanelDisplayOptions_ {
   uint8_t showDemangledFuncName : 1;
   uint8_t showFullPathObject : 1;
} BacktracePanelDisplayOptions;
```

But this brings some padding/performance questions along (most compilers don't optimize these situations well, and even when this oly needs one byte for the actual struct, in practice due to embedding elsewhere you'll likely end up with 4 or 8 bytes anyway).

So nothing saved over the enum variant and compilers are kinda good at handling those in comparison.

Reply:
I would go with a bitfield rather than a "byte field" if the bit flags are related or if the bits are often compared together. Otherwise I just don't know how it would save size when, in assembly, the "bit mask" immediate values would take code size for what you saved by crushing a byte field into a bit field.

I might be overthinking, though.

Reply:
As I said, the enum there is fine. The struct variant (both using bit fields and booleans) produces much more code for access, than the enum variant.

Reply:
If the enum is fine, I prefer to keep it.